template/dockerfiles/gomodule/Dockerfile

```diff
@@ -1,12 +1,14 @@
-FROM golang:{{VERSION}}
+FROM golang:{{VERSION}} AS builder
 ENV PORT {{PORT}}
 EXPOSE {{PORT}}
 
-WORKDIR /go/src/app
+WORKDIR /build
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
 COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -v -o app-binary
 
-RUN go get
-RUN go build -v -o app
-RUN mv ./app /go/bin/
-
-CMD ["app"]
\ No newline at end of file
+FROM alpine
```

Comment:
Any specific reason on why we use alpine? Should we use a distroless image?

Reply:
we can do that I'll swap to something like `gcr.io/distroless/static-debian12` then it'll be even smaller